sdk/batch/Azure.ResourceManager.Batch/tests/TestCase/BatchAccountCertificateOperationTests.cs

```diff
@@ -8,36 +8,59 @@
 using Azure.Core.TestFramework;
 using Azure.ResourceManager.Batch.Models;
 using Azure.ResourceManager.Batch.Tests.Helpers;
+using Azure.ResourceManager.Resources;
 using NUnit.Framework;
 
 namespace Azure.ResourceManager.Batch.Tests.TestCase
 {
     public class BatchAccountCertificateOperationTests : BatchManagementTestBase
     {
+        private ResourceIdentifier _batchAccountId;
+        private BatchAccountResource _batchAccountResource;
+
         public BatchAccountCertificateOperationTests(bool isAsync)
             : base(isAsync)//, RecordedTestMode.Record)
         {
         }
 
-        private async Task<BatchAccountCertificateResource> CreateAccountResourceAsync(string accountName)
+        [OneTimeSetUp]
+        public async Task GlobalSetup()
         {
-            ResourceIdentifier storageAccountId = (await GetStorageAccountResource()).Id;
-            var collection = (await CreateResourceGroupAsync()).GetBatchAccounts();
-            var input = ResourceDataHelper.GetBatchAccountData(storageAccountId);
-            var lro = await collection.CreateOrUpdateAsync(WaitUntil.Completed, Recording.GenerateAssetName("testaccount"), input);
-            var account = lro.Value;
-            var certificateContainer  = account.GetBatchAccountCertificates();
-            var certificateInput = ResourceDataHelper.GetBatchAccountCertificateData();
-            var lroc = await certificateContainer.CreateOrUpdateAsync(WaitUntil.Completed, accountName, certificateInput);
-            return lroc.Value;
+            var rgName = SessionRecording.GenerateAssetName("testrg-batch");
+            var storageAccountName = SessionRecording.GenerateAssetName("azstorageforbatch");
+            var batchAccountName = SessionRecording.GenerateAssetName("testaccount");
+            if (Mode == RecordedTestMode.Playback)
```

Comment:
any chance we can share this setup code in the base class as it seems we have the exact same code in quite a few places?  Will make it less likely that we repeat the mistake when new test classes are made.

Reply:
Agree. An example is https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/CosmosDBManagementClientBase.cs#L31-L38

Reply:
Good catch, updated